### PR TITLE
API updates 2

### PR DESCRIPTION
**Pull request by natefaubion**
This is an alternative to #6. I've removed the privileged "base" effect, and done something which I think is more idiomatic. `aff` and `eff` are just their own effects like any other, and they can be interpreted together if need be.

### CODE REVIEW

**Comment by natefaubion** [comment]
@joneshf @paf31 What do y'all think of this sort of API?

File: src/Run.purs [hunk 1]
```diff
@@ -1,41 +1,43 @@
...
   , expand
-  , BaseEff
-  , BaseAff
+  , EFF
+  , AFF
+  , liftEff
+  , liftAff
+  , runBaseEff
+  , runBaseAff
+  , runBaseAff'
```
Comment (joneshf) [inline src/Run.purs:25]: When i saw this, I thought it was going to do something specific to `Aff a b`. Like require a canceler or some conversion `Run (aff :: AFF e) a -> Run (eff :: EFF (exception :: EXCEPTION | e) (Canceler e)` or something. I think what the function does is fine, but the name is a bit confusing. I don't have a better suggestion though.

**Review by joneshf — COMMENTED** [review]
I like that you don't have to choose between `Aff a b` and `Eff a b` anymore.

File: src/Run.purs (hunk 1 above)
Comment (natefaubion) [reply src/Run.purs:25]: Maybe it should be `toEff` and `toAff`?

**Comment by natefaubion** [comment]
Aaalllrighty then, what about this go around? All the functions now just take handlers from `purescript-variant`. This means multi-interpreters just work out. The only difference is instead of `case_`, you use `send`, which will handle the default case of a variant not matching.

```purescript
let
  multi1 = send
    # on _foo runFoo
    # on _bar runBar
  multi2 = match
    { base1: ...
    , base2: ...
    }
program
  # runWithEffect multi
  # runEffect multi2
```

**Comment by natefaubion** [comment]
I'm happy to consider alternative as well. I don't _love_ the idea of requiring `send`. I can pass it in automatically, at the expensive of longer, more opaque type signatures.

**Pull request merged**